charts/renku/values.yaml

```diff
@@ -70,31 +70,48 @@ keycloak:
   ## Spawn a keycloak instance
   enabled: true
 
-  ## Keycloak admin user configuration
-  username: admin
-  password: admin
+  keycloak:
 
-  ## Keycloak image
-  # image:
-  #   pullPolicy: IfNotPresent
-  #   repository: jboss/keycloak
-  #   tag: 3.4.3.Final
+    ## Keycloak admin user configuration
+    username: admin
+    password: admin
 
-  ## Pod affinity for Keycloak deployment
-  # affinity: {}
-  ## Node selector for Keycloak deployment
-  # nodeSelector: {}
-  ## Pod tolerations for Keycloak deployment
-  # tolerations: []
+    extraVolumes: |
+      - name: realm-secret
+        secret:
+          secretName: renku
+          items:
+          - key: renku-realm.json
+            path: renku-realm.json
 
-  ## Resource requests/limits for Keycloak
-  # resources: {}
+    extraVolumeMounts: |
+      - name: realm-secret
+        mountPath: "/realm/"
+        readOnly: true
+
+    extraArgs: -Dkeycloak.import=/realm/renku-realm.json
+
+    persistence:
+
+      # Disable deployment of the PostgreSQL chart
+      deployPostgres: false
+
+      # The database vendor. Can be either "postgres", "mysql", "mariadb", or "h2"
+      dbVendor: postgres
+
+      ## The following values only apply if "deployPostgres" is set to "false"
+
+      dbName: keycloak
+      dbHost: renku-postgresql
+      dbPort: 5432 # 5432 is PostgreSQL's default port. For MySQL it would be 3306
+      dbUser: keycloak
+
+      existingSecret: renku
```

Comment:
TODO: Document the following helm template caveat (at least in `values.yaml`).

Convention in helm chart is to define name of resources using the following template:
```
{{- define "postgresql.fullname" -}}
{{- if .Values.fullnameOverride -}}
{{- .Values.fullnameOverride | trunc 20 | trimSuffix "-" -}}
{{- else -}}
{{- $name := default .Chart.Name .Values.nameOverride -}}
{{- if contains $name .Release.Name -}}
{{- .Release.Name | trunc 20 | trimSuffix "-" -}}
{{- else -}}
{{- printf "%s-%s" .Release.Name $name | trunc 20 | trimSuffix "-" -}}
{{- end -}}
{{- end -}}
{{- end -}}
```

This makes the resource name depend on the Helm release name.

The values of `dbHost` and `existingSecret` work when the release name is `renku`. As @ebiiii experienced, when deploying to a different release name, then at least `dbHost` and `existingSecret` need to be adjusted.

Reply:
is there no way to define namespace-resolvable aliases that would make this work in general? Like network-specific aliases in `docker-compose`.

Reply:
We could do by using `fullnameOverride`. If we want to have a preference for that, we can investigate it in a separate issue.